test/unitary/xtypes/dynamicdata_operators.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <xtypes/xtypes.hpp>
+#include "../utils.hpp"
+
+using namespace eprosima::xtypes;
+
+#define PI 3.14159f
+
+/***********************************************
+ *         DynamicData Operators Tests         *
+ **********************************************/
+#define ASSERT_EQ_DYNAMICDATA(DYNAMICDATA, TYPE, VAL) ASSERT_EQ(DYNAMICDATA.value<TYPE>(), VAL)
+
+#define ASSERT_EQ_DYNAMICDATA_INT32(DYNAMICDATA, VAL) ASSERT_EQ_DYNAMICDATA(DYNAMICDATA, int32_t, VAL)
+
+TEST (DynamicDataOperators, increment_decrement_operators)
+{
+    DynamicData data(primitive_type<int32_t>());
+    data.value(4);
+    DynamicData& dataref = ++data;
+    ASSERT_EQ_DYNAMICDATA_INT32(dataref, 5);
+    ASSERT_EQ_DYNAMICDATA_INT32(data--, 4);
+}
+
+#define ASSERT_EQ_DYNAMICDATA_ARITHMETIC_OPERATION(OPERATOR, VALUE) \
+{\
+    res = data OPERATOR _data;\
+    ASSERT_EQ(res.value<int32_t>(), VALUE);\
+}
+
+TEST (DynamicDataOperators, arithmetic_operators)
+{
+    DynamicData data(primitive_type<int32_t>());
+    DynamicData _data(primitive_type<int32_t>());
+    DynamicData res(primitive_type<int32_t>());
+    data.value(4);
+    _data.value(-2);
+    ASSERT_EQ_DYNAMICDATA_INT32(-data, -4);
+    ASSERT_EQ_DYNAMICDATA_INT32(~data, -5);
+    ASSERT_EQ_DYNAMICDATA_INT32(!data, false);
+    ASSERT_EQ_DYNAMICDATA_ARITHMETIC_OPERATION(+, 2);
```

Comment:
This is a nice macro but makes difficult to know which is the expected value.
I would modify the macro to receive "data", "_data", something like:

    ASSERT_EQ_DYNAMICDATA_ARITHMETIC_OPERATION(data, +, _data, 2);

This way, is very clear that `data + _data = 2`